frontend/lib/src/components/elements/DeckGlJsonChart/withMapboxToken/withMapboxToken.test.tsx

```diff
@@ -0,0 +1,165 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement } from "react"
+import { customRenderLibContext, render } from "@streamlit/lib/src/test_util"
+import { DeckGlJsonChart as DeckGlJsonChartProto } from "@streamlit/lib/src/proto"
+
+import withMapboxToken, {
+  MapboxTokenFetchingError,
+  TOKENS_URL,
+  WrappedMapboxProps,
+} from "./withMapboxToken"
+import axios from "axios"
+import { screen, waitFor } from "@testing-library/react"
+
+interface TestProps {
+  label: string
+  width: number
+  mapboxToken: string
+}
+
+describe("withMapboxToken", () => {
+  const mockMapboxToken = "mockToken"
+  const element = DeckGlJsonChartProto.create({
+    // mock .streamlit/config.toml token
+    mapboxToken: mockMapboxToken,
+  })
+
+  const emptyElement = DeckGlJsonChartProto.create({})
+
+  function getProps(): WrappedMapboxProps<TestProps> {
+    return {
+      label: "mockLabel",
+      width: 123,
+      element,
+    }
+  }
+
+  jest.mock("axios")
+
+  const MockComponent = (props: {
+    mapboxToken: string | undefined
+  }): ReactElement => (
+    <div data-testid="mock-component">{props.mapboxToken}</div>
+  )
+
+  describe("withMapboxToken rendering", () => {
+    const DeltaType = "testDeltaType"
+    const WrappedComponent = withMapboxToken(DeltaType)(MockComponent)
+    const HOST_CONFIG_TOKEN = "HOST_CONFIG_TOKEN"
+
+    beforeEach(() => {
+      jest.resetAllMocks()
+    })
+
+    it("renders without crashing", async () => {
+      const props = getProps()
+      customRenderLibContext(<WrappedComponent {...props} />, {})
+      const mockComponent = await screen.findByTestId("mock-component")
+      expect(mockComponent.textContent).toBe(mockMapboxToken)
+    })
+
+    it("defines `displayName`", () => {
+      expect(WrappedComponent.displayName).toEqual(
+        "withMapboxToken(MockComponent)"
+      )
+    })
+
+    it("should inject mapbox token to the wrapped component when available in the config", async () => {
+      axios.get = jest.fn().mockImplementation(() => ({
+        data: { userMapboxToken: mockMapboxToken },
+      }))
+
+      customRenderLibContext(
+        <WrappedComponent element={element} width={500} />,
+        {
+          hostConfig: { mapboxToken: HOST_CONFIG_TOKEN },
+        }
+      )
+
+      await waitFor(() => {
+        const element = screen.getByTestId("mock-component")
+        expect(element.textContent).toBe(mockMapboxToken)
```

Comment:
nit: I think this way of testing if the correct token was supplied feels a bit hacky. It's a bit confusing since in the actual implementation, the token isn't part of the HTML text content. Maybe a better way might be to not use the screen aspects of testing that, but instead just compare the value from the state of the component. Or maybe even mock setState and check if it is called with the correct token...

Reply:
If we keep it as is, maybe a comment to the `MockComponent` that it is only used to test that the mapbox token is correctly set.